File: mixin-padang-garut/src/main/java/com/andia/mixin/padang/garut/converters/TableConverter.java
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.andia.mixin.padang.garut.converters;

import com.andia.mixin.padang.garut.DataminerTable;
import com.andia.mixin.padang.garut.DataminerValue;
import com.andia.mixin.padang.garut.values.GarutTable;
import com.google.protobuf.AbstractMessage;

public class TableConverter implements ValueConverter {

	private static TableConverter instance;

	private TableConverter() {

	}

	public static TableConverter getInstance() {
		if (instance == null) {
			instance = new TableConverter();
		}
		return instance;
	}

	@Override
	public Object fromValue(DataminerValue value) {
		DataminerTable table = value.getTable();
		return createTable(table);
	}

	private GarutTable createTable(DataminerTable table) {
		return new GarutTable(table);
	}

	@Override
	public Object fromProtobuf(AbstractMessage value) {
		DataminerTable object = (DataminerTable) value;
		return createTable(object);
	}
}

File: mixin-padang-vegazoo/src/main/js/vegazoo/directors/converters/TableConverter.ts
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>. */
import VisageConstant from "bekasi/visage/VisageConstant";
import VisageTable from "bekasi/visage/VisageTable";

export default class TableConverter {

	public convertToJsonspec(table: VisageTable): any {

		// Baca dataset
		let result: any[] = [];
		for (let i = 0; i < table.recordCount(); i++) {
			let columns = table.getColumns();
			let record = table.getRecord(i);
			let values = record.getValues();
			let object: any = {};
			for (let j = 0; j < columns.length; j++) {
				let column = columns[j];
				let key = column.getKey();
				let value = values[j];
				if (value instanceof VisageConstant) {
					object[key] = value.getValue();
				} else {
					object[key] = value.toString();
				}
			}
			result.push(object);
		}

		return result;

	}

}
